pyrobosim/pyrobosim/navigation/astar_grid.py

```diff
@@ -0,0 +1,264 @@
+    """Occupancy grid based A* planner"""
+
+    def __init__(
+        self,
+        world,
+        resolution=0.05,
+        inflation_radius=0.0,
+        distance_metric="Manhattan",
+        diagonal_motion=True,
+        max_time=5.0,
+    ) -> None:
+        """
+        Creates a grid based A* planner
+
+        :param world: World object to use in the planner.
+        :type world: :class:`pyrobosim.core.world.World`
+        :param resolution: The resolution to be used in the occupancy grid
+        :type resolution: float
+        :param inflation_radius: The inflation radius to be used for the planner's occupancy grid
+        :type inflation_radius: float
+        :param distance_metric: The metric to be used as heuristic
+        :type distance_metric: string 'Manhattan' or 'Euclidian'
+        :param diagonal_motion: If to use expand nodes using diagonal motion
+        :type diagonal_motion: bool
+        :param max_time: Maximum time allowed for planning
+        :type max_time: float
+        """
+        self.grid = occupancy_grid_from_world(
+            world, resolution=resolution, inflation_radius=inflation_radius
+        )
+        self.max_time = max_time
+        self.world = world
+        self.resolution = resolution
+        self.inflation_radius = inflation_radius
+        self.candidates = PriorityQueue()  # Node
+        self.visited = set()  # (x, y)
+        self.start = None
+        self.goal = None
+        self.num_nodes_expanded = 0
+        self.distance_metric = distance_metric
+        self.diagonal_motion = diagonal_motion
+        self.actions = {
+            "left": {"cost": 1, "action": (-1, 0)},
+            "right": {"cost": 1, "action": (1, 0)},
+            "up": {"cost": 1, "action": (0, 1)},
+            "down": {"cost": 1, "action": (0, -1)},
+            "left_up": {"cost": 1.414, "action": (-1, 1)},
+            "left_down": {"cost": 1.414, "action": (-1, -1)},
+            "right_up": {"cost": 1.414, "action": (1, 1)},
+            "right_down": {"cost": 1.414, "action": (1, -1)},
+        }
+        keys = list(self.actions.keys())
+        self.selected_actions = keys if self.diagonal_motion else keys[:4]
+        print(f"Selected actions : {self.selected_actions}")
+
+        # Set the heuristic function
+        if self.distance_metric == "Manhattan":
+            self._heuristic = lambda p1, p2: abs(p1[0] - p2[0]) + abs(p1[1] - p2[1])
+        elif self.distance_metric == "Euclidian":
+            # Not taking the square root since we dont need the actual distance
+            self._heuristic = lambda p1, p2: (p1[0] - p2[0]) ** 2 + (p1[1] - p2[1]) ** 2
+        print(f"Selected heuristic: {self.distance_metric}")
+
+    def _reset(self):
+        """Resets the state of data used by the algorithm"""
+        self.grid = occupancy_grid_from_world(
+            self.world,
+            resolution=self.resolution,
+            inflation_radius=self.inflation_radius,
+        )
```

Comment:
Oh, this can be removed, at the time I put this here, my thinking was that we would be making a new grid with the other robot positions included as obstacles every time plan() was called

Reply:
I think it's OK to ignore other robots -- in practice, if there are other robots they almost always would not be static obstacles, so would be ignored by a global planner like this one.